pkg/telemetrylogs/key_evolution_metadata.go

```diff
@@ -0,0 +1,163 @@
+	return &KeyEvolutionMetadata{
+		cache:          cache,
+		telemetryStore: telemetryStore,
+		logger:         logger,
+	}
+}
+
+func (k *KeyEvolutionMetadata) fetchFromClickHouse(ctx context.Context, orgID valuer.UUID) {
+	store := k.telemetryStore
+	logger := k.logger
+
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	if store.ClickhouseDB() == nil {
+		logger.WarnContext(ctx, "ClickHouse connection not available for key evolution metadata fetch")
+		return
+	}
+
+	// Build query to fetch all key evolution metadata
+	sb := sqlbuilder.NewSelectBuilder()
+	sb.Select(
+		"base_column",
+		"base_column_type",
+		"new_column",
+		"new_column_type",
+		"release_time",
+	)
+	sb.From(fmt.Sprintf("%s.%s", KeyEvolutionMetadataDBName, KeyEvolutionMetadataTableName))
+	sb.OrderBy("base_column", "release_time")
+
+	query, args := sb.BuildWithFlavor(sqlbuilder.ClickHouse)
+
+	rows, err := store.ClickhouseDB().Query(ctx, query, args...)
+	if err != nil {
+		logger.WarnContext(ctx, "Failed to fetch key evolution metadata from ClickHouse", "error", err)
+		return
+	}
+	defer rows.Close()
+
+	// Group metadata by base_column
+	metadataByKey := make(map[string][]*telemetrytypes.KeyEvolutionMetadataKey)
+
+	for rows.Next() {
+		var (
+			baseColumn     string
+			baseColumnType string
+			newColumn      string
+			newColumnType  string
+			releaseTime    uint64
+		)
+
+		if err := rows.Scan(&baseColumn, &baseColumnType, &newColumn, &newColumnType, &releaseTime); err != nil {
+			logger.WarnContext(ctx, "Failed to scan key evolution metadata row", "error", err)
+			continue
+		}
+
+		key := &telemetrytypes.KeyEvolutionMetadataKey{
+			BaseColumn:     baseColumn,
+			BaseColumnType: baseColumnType,
+			NewColumn:      newColumn,
+			NewColumnType:  newColumnType,
+			ReleaseTime:    time.Unix(0, int64(releaseTime)),
+		}
+
+		metadataByKey[baseColumn] = append(metadataByKey[baseColumn], key)
+	}
+
+	if err := rows.Err(); err != nil {
+		logger.WarnContext(ctx, "Error iterating key evolution metadata rows", "error", err)
+		return
+	}
+
+	// Store each key's metadata in cache
+	for keyName, keys := range metadataByKey {
+		cacheKey := KeyEvolutionMetadataCacheKeyPrefix + keyName
+		cachedData := &CachedKeyEvolutionMetadata{Keys: keys}
+		if err := k.cache.Set(ctx, orgID, cacheKey, cachedData, 24*time.Hour); err != nil {
+			logger.WarnContext(ctx, "Failed to set key evolution metadata in cache", "key", keyName, "error", err)
+		}
+	}
+
+	logger.DebugContext(ctx, "Successfully fetched key evolution metadata from ClickHouse", "count", len(metadataByKey))
+}
+
+// Add adds a metadata key for the given key name and orgId.
+// This is primarily for testing purposes. In production, data should come from ClickHouse.
+func (k *KeyEvolutionMetadata) Add(ctx context.Context, orgId valuer.UUID, keyName string, key *telemetrytypes.KeyEvolutionMetadataKey) {
+	k.logger.WarnContext(ctx, "Add is not implemented for key evolution metadata")
+
+}
```

Comment:
Let's not do this. Since you already have the implementation for tests that implements the interface, you can populate the required data during the creation of new mock store rather adding `Add` to interface just to aid in testing.